source/dshow-dialogbox.cpp

```diff
@@ -0,0 +1,50 @@
+#include "dshow-dialogbox.hpp"
+#include "log.hpp"
+
+namespace DShow {
+    DeviceDialogBox::DeviceDialogBox() :
+        deviceFilter(nullptr),
+        threadId(0),
+        threadHandle(nullptr),
+        isOpen(false)
+    {
+    }
+
+    void DeviceDialogBox::Open(IUnknown* filter) {
+        if (!isOpen) {
+            deviceFilter = filter;
+            threadHandle = CreateThread(nullptr, 0, CallCreate, (void*) this, 0, &threadId);
+
+            if (threadHandle == nullptr) {
+                Warning(L"Could not create thread for filter dialog box");
+            }
+        }
+    }
+
+    void DeviceDialogBox::Close() {
+        if (isOpen) {
+            PostThreadMessage(threadId, WM_QUIT, 0, 0);
+            CloseHandle(threadHandle);
```

Comment:
Shouldn't we wait until the thread terminates before closing the handle using the `WaitForSingleObject()` function?

Reply:
I don't have control over the dialog box window message loop. So I need to force close the thread with WM_QUIT in order to close a dialog box that is still open when we are closing SLOBS. It would need a call for WaitForSingleObject if I used WM_CLOSE instead of WM_QUIT.

Reply:
Got it